src/AutoRest.CSharp/Common/Output/Models/ValueExpressions.cs

```diff
@@ -0,0 +1,291 @@
+                    };
+
+                    return new StaticMethodCallExpression(typeof(Azure.Core.ProtocolOperationHelpers), methodName, arguments, false, async);
+                }
+
+                public static ValueExpression Convert(CSharpType responseType, CodeWriterDeclaration response, CodeWriterDeclaration clientDiagnostics, string scopeName)
+                {
+                    var responseVariable = new VariableReference(response);
+                    var fromResponseReference = new MemberReference(new TypeReference(responseType), "FromResponse");
+                    var diagnosticsReference = new VariableReference(clientDiagnostics);
+                    var arguments = new ValueExpression[] { responseVariable, fromResponseReference, diagnosticsReference, new FormattableStringToExpression($"{scopeName:L}") };
+                    return new StaticMethodCallExpression(typeof(Azure.Core.ProtocolOperationHelpers), nameof(Azure.Core.ProtocolOperationHelpers.Convert), arguments);
+                }
+            }
+
+            public static class Response
+            {
+                public static ValueExpression FromValue(CSharpType responseType, CodeWriterDeclaration response)
+                {
+                    var responseVariable = new VariableReference(response);
+                    var fromResponseCall = new StaticMethodCallExpression(responseType, "FromResponse", new[]{responseVariable}, false, false);
+                    return new StaticMethodCallExpression(typeof(Azure.Response), nameof(Azure.Response.FromValue), new ValueExpression[]{fromResponseCall, responseVariable});
+                }
+            }
+        }
+    }
+
+    internal record ValueExpression
+    {
+        public static implicit operator ValueExpression(Parameter parameter) => new ParameterReference(parameter);
+        public static implicit operator ValueExpression(CodeWriterDeclaration name) => new VariableReference(name);
+    }
+
+    internal record DefaultValueExpression(CSharpType Type) : ValueExpression;
+    internal record NewInstanceExpression(CSharpType Type, IReadOnlyList<ValueExpression> Arguments) : ValueExpression
+    {
+        public IReadOnlyDictionary<string, ValueExpression> Properties { get; init; } = new Dictionary<string, ValueExpression>();
+    }
+
+    internal record TernaryConditionalOperator(ValueExpression Condition, ValueExpression Consequent, ValueExpression Alternative) : ValueExpression;
+    internal record TypeReference(CSharpType Type) : ValueExpression;
+    internal record MemberReference(ValueExpression Inner, string MemberName) : ValueExpression;
+    internal record ParameterReference(Parameter Parameter) : ValueExpression;
+    internal record VariableReference(CodeWriterDeclaration Name) : ValueExpression;
+    internal record FormattableStringToExpression(FormattableString Value) : ValueExpression; // Shim between formattable strings and expressions
```

Comment:
I believe the purpose of this record here is for compatibility only, but I think we should not abuse this class.
The purpose of this PR is trying to give our writers "semantic meanings" so that we could more likely to write code that could compile and maybe more easy to read. But this class is kind of following the "old ways" - I do not have a meaning, just write whatever I have here.

Therefore I suggest for everywhere we are using the `FormattableStringToExpression` record, we might need to take a look and see if we could use an existing static method or come up to a static method to describe that usage, and eventually remove this.